client/my-sites/plans/jetpack-plans/jetpack-social-free-card/index.tsx

```diff
@@ -0,0 +1,88 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { TERM_ANNUALLY } from '@automattic/calypso-products';
+import classNames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import { useCallback, useMemo } from 'react';
+import { useDispatch } from 'react-redux';
+import JetpackProductCard from 'calypso/components/jetpack/card/jetpack-product-card';
+import isJetpackCloud from 'calypso/lib/jetpack/is-jetpack-cloud';
+import { ITEM_TYPE_PRODUCT } from 'calypso/my-sites/plans/jetpack-plans/constants';
+import type { QueryArgs, SelectorProduct } from 'calypso/my-sites/plans/jetpack-plans/types';
+
+const SOCIAL_FREE_URL = isJetpackCloud()
+	? 'https://cloud.jetpack.com/pricing/jetpack-social/welcome'
+	: 'https://wordpress.org/plugins/jetpack-social/'; // This may need to be updated (page doesn't exist yet).
+
+const useSocialFreeItem = (): SelectorProduct => {
+	const translate = useTranslate();
+
+	return useMemo(
+		() => ( {
+			productSlug: 'jetpack-social-free',
+			isFree: true,
+			displayName: translate( 'Social' ),
+			features: {
+				items: [
+					{ slug: 'not used', text: translate( 'Schedule or share instantly' ) },
+					{ slug: 'not used', text: translate( 'Connect to leading social platforms' ) },
+					{ slug: 'not used', text: translate( 'Manage everything from Publishing Hub' ) },
+				],
+			},
+			type: ITEM_TYPE_PRODUCT, // not used
+			term: TERM_ANNUALLY, // not used
+			iconSlug: 'not used',
+			shortName: 'not used',
+			tagline: 'not used',
+			description: 'not used',
+		} ),
+		[ translate ]
+	);
+};
+
+export type CardWithPriceProps = {
+	siteId: number | null;
+	urlQueryArgs: QueryArgs;
+};
+
+const CardWithPrice: React.FC< CardWithPriceProps > = ( { siteId, urlQueryArgs } ) => {
+	const translate = useTranslate();
+	const socialFreeProduct = useSocialFreeItem();
+
+	const dispatch = useDispatch();
+	const trackCallback = useCallback(
+		() =>
+			dispatch(
+				recordTracksEvent( 'calypso_product_jpsocialfree_click', {
+					site_id: siteId ?? undefined,
+				} )
+			),
+		[ dispatch, siteId ]
+	);
+
+	const onButtonClick = useCallback( () => {
```

Comment:
This seems redundant, doesn't it? The `onButtonClick` callback appears to only be calling another function, with no additional state or logic. What if instead we remove this variable entirely and change the below `onButtonClick` property to point at the `trackCallback` method?

Reply:
Yes definitely, thanks Matt!  Fixed.